src/interfaces/business-i.ts

```diff
@@ -1,13 +1,98 @@
+import type { CorpTypeCd, AmalgamationTypes, FilingTypes } from '@bcrs-shared-components/enums'
+import type { ApiDateTimeUtc, IsoDatePacific } from '@bcrs-shared-components/interfaces'
+
 import { BusinessStateE } from '~/enums/business-state-e'
-import { BusinessTypeE } from '~/enums/business-type-e'
+import { FilingSubTypeE } from '~/enums/filing-sub-type-e'
+import type { StateFilingHeaderI } from '~/interfaces/state-filing-i'
+import type { WarningTypesE } from '~/enums/warning-types-e'
 
-export interface BusinessI {
+export interface SlimBusinessI {
   adminFreeze: boolean
   alternateNames: { operatingName: string }[]
   goodStanding: boolean
   identifier: string
   legalName: string
-  legalType: BusinessTypeE
+  legalType: CorpTypeCd
   state: BusinessStateE
   taxId?: string
+  inDissolution: boolean
+}
+
+export interface StateFilingI {
+  business: any
+  consentContinuationOut?: any
+  header: StateFilingHeaderI
+  dissolution?: any
+  restoration?: any
+  putBackOn?: any
+}
+
+export interface FilingTypeI {
+  displayName: string
+  feeCode: string
+  name: FilingTypes
+  type?: FilingSubTypeE
+}
+
+export interface AllowedActionsI {
+  digitalBusinessCard: boolean
+  filing: {
+    filingSubmissionLink: string
+    filingTypes: FilingTypeI[]
+  }
+}
+
+export interface AmalgamatedIntoI {
+  amalgamationDate: ApiDateTimeUtc
+  amalgamationType: AmalgamationTypes
+  identifier: string // eg, BC7654321
+  legalName: string
+}
+
+export interface BusinessWarningI {
+  code: string // FUTURE: use an enum
+  filing?: string // not used
+  message: string
+  warningType: WarningTypesE
+  data?: any // optional extra properties (eg, amalgamationDate)
+}
+
+// comments come from business-filings-ui project ApiBusinessIF interface
+export interface BusinessI {
```

Comment:
Could we extend the 'slim' interface?

Reply:
Did that too, sometimes I forget that this is option in JS :P